pkg/utils/transform/transform.go

```diff
@@ -69,12 +69,30 @@ func InjectNamespaceConditional(preserveNamespace, targetNamespace string) mf.Tr
 	}
 }
 
+func InjectNamespaceRoleBindingConditional(preserveNS, preserveRBSubjectNS, targetNamespace string) mf.Transformer {
+	tf := InjectNamespaceRoleBindingSubjects(targetNamespace)
+
+	return func(u *unstructured.Unstructured) error {
+		annotations := u.GetAnnotations()
+		val, ok := annotations[preserveNS]
+		if !(ok && val == "true") {
+			u.SetNamespace(targetNamespace)
+		}
+		val, ok = annotations[preserveRBSubjectNS]
+		if ok && val == "true" {
+			return nil
+		}
+		return tf(u)
+	}
+}
+
 func InjectNamespaceRoleBindingSubjects(targetNamespace string) mf.Transformer {
 	return func(u *unstructured.Unstructured) error {
 		kind := strings.ToLower(u.GetKind())
 		if kind != "rolebinding" {
 			return nil
 		}
+		u.SetNamespace(targetNamespace)
```

Comment:
this is a shortcut, and could be confusing. here we are making a function with some other resposibility one additional thing. let us try to handle this differently.

In addition, why don't we keep `transform.InjectNamespaceRoleBindingSubjects(cfg.Spec.TargetNamespace),` in rbTransforms as well.

Reply:
That's there by mistake. Deleting it.